src/plugins/responsive-preview/index.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * Internal dependencies
+ */
+import './auto-change-responsive-preview'
+
+/**
+ * External dependencies
+ */
+import {
+	keys, inRange, throttle,
+} from 'lodash'
+
+/**
+ * Wordpress dependencies
+ */
+import domReady from '@wordpress/dom-ready'
+import { addAction } from '@wordpress/hooks'
+import { select } from '@wordpress/data'
+
+const query = '.edit-post-visual-editor'
+
+/**
+ * CSS files to cache
+ *
+ * It includes Stackable custom css files and inline styles for blocks.
+ */
+const includesCss = [
+	'/dist/frontend_blocks__premium_only.css',
+	'/dist/frontend_blocks.css',
+	'/dist/editor_blocks__premium_only.css',
+	'/dist/editor_blocks.css',
+]
+
+const cssObject = {}
+
+// The previous mode.
+let previousMode = 'Desktop'
+
+// Cache the responsive widths of the preview.
+const widthsDetected = {}
+
+/**
+ * Populates the cssObject with media queries.
+ */
+export const cacheCssObject = () => {
+	// Stores the indices of needed styleSheets
+	const stylesheetIndices = []
+
+	const styleSheets = Array.from( document.styleSheets )
+
+	styleSheets.forEach( ( { href }, index ) => {
+		// Only do this for our own css files.
+		if ( href && includesCss.some( url => href.includes( url ) ) ) {
+			stylesheetIndices.push( index )
+		// Also do this for style tags.
+		} else if ( href === null ) {
+			stylesheetIndices.push( index )
+		}
+	} )
+
+	stylesheetIndices.forEach( index => {
+		const mediaIndices = {}
+
+		if ( styleSheets[ index ].cssRules ) {
+			Array.from( styleSheets[ index ].cssRules ).forEach( ( { cssText, media }, mediaIndex ) => {
+				if ( media && cssText.includes( '.ugb' ) && media.mediaText.match( /(max|min)-width/ ) ) {
+					const maxWidth = media.mediaText.match( /max-width:\s*(\d+)px/ )
+					const minWidth = media.mediaText.match( /min-width:\s*(\d+)px/ )
+
+					const max = maxWidth ? parseInt( maxWidth[ 1 ], 10 ) : 9999
+					const min = minWidth ? parseInt( minWidth[ 1 ], 10 ) : 0
+
+					if ( cssObject && cssObject[ index ] && cssObject[ index ][ mediaIndex ] ) {
+						const {
+							min: currentMin, max: currentMax,
+						} = cssObject[ index ][ mediaIndex ]
+
+						if ( min === 5000 || max === 5000 ) {
+							// Store the cached value of media query has already been modified.
+							mediaIndices[ mediaIndex ] = { ...cssObject[ index ][ mediaIndex ] }
+						} else if ( currentMin !== min || currentMax !== max ) {
+							// Store the new media query if custom CSS is modified.
+							mediaIndices[ mediaIndex ] = {
+								mediaText: media.mediaText,
+								min,
+								max,
+								cssText,
+							}
+						} else {
+							// Store the cached value if nothing is modified.
+							mediaIndices[ mediaIndex ] = { ...cssObject[ index ][ mediaIndex ] }
+						}
+					} else {
+					// Store the new media query if the value is not found in cached media queries.
+					 mediaIndices[ mediaIndex ] = {
+							mediaText: media.mediaText,
+							min,
+							max,
+							cssText,
+					 }
+					}
```

Comment:
I'm a bit uncomfortable in comparing a number for the min and max here (even though it's almost unlikely that someone will use 5000px).